documentation.md

```diff
@@ -0,0 +1,96 @@
+# Typology Processor
+
+## Overview
+
+Here's a brief explanation of each participant:
+
+- Caller: The caller initiates the request to process a transaction.
+- Handle Transaction: This function handles the transaction, coordinates the processing steps, and accumulates the results.
+- Execute Request: This function processes the transaction for each typology and sends the results to CMS and CADP.
+- Evaluate Typology Expr: This function computes the typology result value based on the provided expression.
+- Redis Cache: Redis Cache stores interim rule results for faster processing and manages cache keys.
+- Database: The Database provides the typology expression needed for evaluation.
+- CMS: The CMS receives typology results that exceed the threshold.
+- CADP: The CADP receives all the typology results.
+
+## Sequence Diagram
+
+```mermaid
+sequenceDiagram
+    autonumber
+    participant Caller as Caller
+    participant HandleTransaction as Handle Transaction
+    participant ExecuteRequest as Execute Request
+    participant EvaluateTypologyExpr as Evaluate Typology Expr
+    participant RedisCache as Redis Cache
+    participant Database as Database
+    participant CMS as CMS
+    participant CADP as CADP
+
+    Note over Caller, HandleTransaction: The process starts by handling a transaction.
+
+    Caller->>HandleTransaction: call(transaction, networkMap, ruleResult)
+    HandleTransaction->>+ExecuteRequest: call(transaction, typology, ruleResult, networkMap)
+
+    Note over ExecuteRequest: Execute Request processes the transaction for each typology.
+
+    loop channel.typologies
```

Comment:
The loop should not be run for ALL the typologies in the (a) channel, but there should be an explicit determination of only the typologies that contain the incoming rule. Please confirm if this is the case, or if the typology processor simply runs over ALL the typologies in the network map (this would be incorrect)? And also, if the loop will cover all the channels, since the `channel[]` is (should be) an array.

Reply:
Filters for typologies in the channel first where there is a rule in the networkmap with the same ID as the incoming rule result. 
https://github.com/frmscoe/typology-processor/blob/d3a181d5d584e5e4d81b2618c434a3c54b545a15/src/logic.service.ts#LL173C91-L173C91